src/patternfly/components/InlineEdit/inline-edit.scss

```diff
@@ -0,0 +1,206 @@
+// Don't remove this magic comment. See gulpfile.js.
+// @import "../../sass-utilities/all";
+
+.pf-c-inline-edit {
+  // Group
+  --pf-c-inline-edit__group--item--MarginRight: var(--pf-global--spacer--sm);
+
+  // Value group
+  --pf-c-inline-edit__group--m-value-group--item--MarginRight: var(--pf-global--spacer--sm);
+
+  // Input group
+  --pf-c-inline-edit__group--m-input-group--item--MarginRight: var(--pf-global--spacer--sm);
+
+  // Action group
+  --pf-c-inline-edit__group--m-action-group--item--MarginRight: var(--pf-global--spacer--md);
+  --pf-c-inline-edit__group--m-input-group--m-action-group--item--MarginRight: 0;
+
+  // Valid action
+  --pf-c-inline-edit__action--m-valid--c-button--m-plain--Color: var(--pf-global--link--Color);
+  --pf-c-inline-edit__action--m-valid--c-button--m-plain--hover--Color: var(--pf-global--link--Color--hover);
+
+  // Icon group
+  --pf-c-inline-edit__group--m-icon-group--item--MarginRight: 0;
+
+  // Toggle
+  --pf-c-inline-edit--m-active__toggle--Color: var(--pf-global--Color--100);
+
+  // Footer group
+  --pf-c-inline-edit__group--m-footer--MarginTop: var(--pf-global--spacer--xl);
+
+  // Hidden
+  .pf-m-hidden {
+    display: none;
+    visibility: hidden;
+  }
+
+  .pf-m-display-inline-flex {
+    display: inline-flex;
+  }
+
+  // Align right
+  .pf-m-align-right {
+    margin-left: auto;
+  }
```

Comment:
while these modifiers _should_ do the same thing between components, we can't always rely on that being the case. since `.pf-m-inline-edit` can wrap a bunch of things, can you prefix the modifiers with the selectors they apply to so they don't interfere with whatever else might be a descendent of `.pf-m-inline-edit`?

Reply:
One thing I noticed that's sorta on this topic is that the modifier `pf-m-error` doesn't receive it's full style properties when inside of a `pf-c-inline-edit` as it's scoped to `pf-c-form` with `--pf-c-form--m-error--Color`. Is there something we can do to make this work? I can add pf-c-form to the same element as pf-c-inline-edit but probably better handled on the CSS side.. wdyt?

Reply:
Maybe it's a matter of making `pf-m-error` more generic? Otherwise, the error text doesn't receive the red error color. Here's the markup I'm using as an immediate sibling to the TextInput react component

`<div class="pf-c-form__helper-text pf-m-error" aria-live="polite">This field is required</div>`

Is there more appropriate html to use? 

<img width="1272" alt="Screen Shot 2020-01-16 at 11 54 41 AM" src="https://user-images.githubusercontent.com/5942899/72644761-d788a180-393f-11ea-80a9-86b4986aa454.png">

I can raise an issue if need be, just wanna make sure I understand what exactly to ask for.

Reply:
Raised this issue to capture the problem better - feel free to comment/reword as necessary! https://github.com/patternfly/patternfly-next/issues/2588

Reply:
> while these modifiers should do the same thing between components, we can't always rely on that being the case. since .pf-m-inline-edit can wrap a bunch of things, can you prefix the modifiers with the selectors they apply to so they don't interfere with whatever else might be a descendent of .pf-m-inline-edit?

It's pretty essential that we allow `*` descendants to be editable, regardless of type for a couple of reasons.

Inline edit can be self-contained or apply to a larger component. To scope `.pf-m-editable` to specific elements that could apply would be to scope it to all elements within all components. So scoping to anything within `pf-c-inline-edit` is intentional. Examples would making a `table row` editable. We can't add a wrapping element around a `table row`, so we have to apply `-editable` to `<tr>` add `-inline-edit`s components to `<td>`s. Same thing with `<dl>`s.  To consider each use case like that could inefficient bearing in mind that this is an invisible, utility component.

Reply:
It looks like you commented out a couple of the problematic classes except for `.pf-m-hidden` and the `.pf-m-grid [data-attr]` selector. If we don't bind those modifiers to an element, we just run the risk of them applying inadvertently to nested things we don't intend on. Would it be possible to wrap whatever you apply `.pf-m-hidden` to in an inline-edit element and apply `display: contents;` to it, then we could scope the modifier class to that?

We could always refactor patternfly to use proper BEM modifier classes so we don't have this problem :)